_posts/2022-09-12-stargate-v2-beta.md

```diff
@@ -0,0 +1,77 @@
+---
+layout: post
+title: "Stargate v2 Beta: Open for extension"
+date:   2022-09-11 00:00:00 -0600
+synopsis: "An update on Stargate v2 including announcement of the first Beta release"
+author_info:
+  name:
+    first: "Mark"
+    last: "Stone"
+  picture: "assets/images/stargate-profile.png"
+---
+
+Last Friday, we released the first [Beta](https://github.com/stargate/stargate/releases/tag/v2.0.0-BETA-1) for Stargate v2. In this post we’ll look at why this is a major milestone for the project and how you can try this new release out for yourself.
+
+## What is Stargate v2?
+
+To answer this question, let’s start with a quick refresher on the Stargate project. We like to refer to Stargate as a “data API gateway” (or data gateway for short) built on top of Apache Cassandra.
+
+In a technology world that is increasingly dominated by cloud data services rather than databases, application developers are likely to think in terms of data abstractions like JSON rather than an idiom unique to a particular database. An API gateway is therefore an ideal way to meet developers where they are, allowing them to work in frameworks and structures familiar to them without having to know all the particulars of the underlying database. Stargate is designed to meet this need – backed by the power of Cassandra, and presenting REST, Document, and GraphQL as developer friendly APIs. We’ve also added a set of gRPC libraries for doing CQL over gRPC as an easier, lightweight, and more cloud friendly alternative to native drivers for CQL.
+
+Stargate v2 represents a significant architectural update for our data API gateway, designed to address several opportunities for improvement we identified in working with the original architecture. In particular:
+
+* Monolith to modular. The original Stargate codebase is too tightly coupled, making it hard to develop anything without knowing everything. Going forward, we’re factoring out the REST, GraphQL and Document APIs as independent services. Enhancing a particular API service, or adding an entirely new API service, will be much easier, while CQL and gRPC APIs remain in the coordinator node for optimal performance.
```

Comment:
```suggestion
* Monolith to modular. The original Stargate codebase is too tightly coupled to the persistence engine, making it hard to develop anything without knowing everything. Going forward, we’re factoring out the REST, GraphQL and Document APIs as independent services. Enhancing a particular API service, or adding an entirely new API service, will be much easier, while CQL and gRPC APIs remain in the coordinator node for optimal performance.
```

Reply:
Or whatever you want to put there. But the phrase "is too tightly coupled" must have "with what?"